core/capabilities/gateway_connector/connector_handler.go

```diff
@@ -0,0 +1,65 @@
+package gateway_connector
+
+import (
+	"context"
+	"crypto/ecdsa"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+
+	"github.com/smartcontractkit/chainlink/v2/core/config"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/api"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/common"
+	"github.com/smartcontractkit/chainlink/v2/core/services/gateway/connector"
+)
+
+type workflowConnectorHandler struct {
```

Comment:
Let's rename it to connectorSigner, strip down as much as we can and put it together with the Wrapper in the same file. Soon the handler portion will go away as I explained in the other comment below.